File: client-samples/mail/java/client/mail/test/MessageListTest.java
/*
 * © Copyright IBM Corp. 2017
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); 
 * you may not use this file except in compliance with the License. 
 * You may obtain a copy of the License at:
 * 
 * http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software 
 * distributed under the License is distributed on an "AS IS" BASIS, 
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
 * implied. See the License for the specific language governing 
 * permissions and limitations under the License.
 */

package client.mail.test;

import io.swagger.client.ApiClient;
import io.swagger.client.ApiException;
import io.swagger.client.api.MessageListApi;
import io.swagger.client.model.MessageListEntry;
import io.swagger.client.model.MessageListResponse;
import io.swagger.client.model.Person;

public class MessageListTest {

    private MessageListApi _api;

    public MessageListTest(String basePath, String user, String password) {
        _api = new MessageListApi();
        ApiClient client = _api.getApiClient();
        client.setBasePath(basePath);
        client.setUsername(user);
        client.setPassword(password);
    }

    public static void main(String[] args) {

        // IMPORTANT: Change these values to match your test environment:
        //
        // - basePath is the base URL for your Domino server.
        //   The mail service MUST be installed and enabled
        //   on the server. Be sure to change the protocol to
        //   https if necessary.
        //
        // - folder is mail file database folder.
        //
        // - database is the mail file database name.
        //
        // - user is the user name of the owner of the mail file specified by
        //   /{folder}/{database}.
        //
        // - password is the user's password.
        //
        String basePath = "http://yourserver.yourorg.com";
        String folder = "mail";
        String database = "database.nsf";
        String user = "First Last";
        String password = "password";

        // Create an instance of the test class
        MessageListTest test = new MessageListTest(basePath, user, password);

        // Read the inbox
        test.readInbox(folder, database);
    }

    /**
     * Reads the inbox and dumps it's messages to the console.
     * 
     * @param folder
     * @param database
     */
    private void readInbox(String folder, String database) {
        try {

            // Read the inbox
            System.out.println("Reading inbox from " + folder + "/" + database + " ...");
            MessageListResponse result = _api.folderDatabaseApiMailInboxGet(folder, database, 
                                            null,           // start Specifies the starting entry.
                                            5,              // count Specifies the number of entries to return.
                                            null,           // page Page number.
                                            null,           // ps Page size or the number of entries to return.
                                            null,           // search Returns only documents that match a full-text query.
                                            null,           // searchmaxdocs Limits the number of documents returned by a full-text search.
                                            "date",         // sortcolumn Returns entries sorted on a column.
                                            "descending",   // sortorder Specifies the sort order.
                                            null,           // keys Returns entries whose initial characters match keys.
                                            null            // keysexactmatch Returns entries that match keys exactly.
                                            );

            // Does the response have messages?
            if ( result != null && result.size() > 0 ) {

                // Dump the messages
                System.out.println("Read inbox request succeeded.  Response follows ...\n");
                for (int i = 0; i < result.size(); i++) {
                    MessageListEntry message = result.get(i);
                    System.out.println("Message " + i);
                    System.out.println("  subject: " + message.getSubject());
                    System.out.println("  date: " + message.getDate());
                    System.out.println("  href: " + message.getHref());
                    
                    Person person = message.getFrom();
                    if ( person != null ) {
                        System.out.println("  from: " + person.getDisplayName());
                    }
                    
                    System.out.println();
                }
            }
            else {

                // Unexpected response
                System.out.println("Read inbox request succeeded, but the response doesn't include a list of messages.");
            }
        }
        catch (ApiException e) {
            System.err.println("Exception when calling MessageListApi#folderDatabaseApiMailInboxGet");
            String body = e.getResponseBody();
            if (body != null) {
                System.err.println("Response from server ...");
                System.err.println(body);
            }
            else {
                e.printStackTrace();
            }
        }
    }

}

File: client-samples/mail/nodejs/messagelisttest.js
/*
 * © Copyright IBM Corp. 2017
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); 
 * you may not use this file except in compliance with the License. 
 * You may obtain a copy of the License at:
 * 
 * http://www.apache.org/licenses/LICENSE-2.0 
 * 
 * Unless required by applicable law or agreed to in writing, software 
 * distributed under the License is distributed on an "AS IS" BASIS, 
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or 
 * implied. See the License for the specific language governing 
 * permissions and limitations under the License.
 */

// IMPORTANT: Change the required package name if necessary.

var api = require('ibm_domino_mail_api');

// IMPORTANT: Change these values to match your test environment:
//
// - basePath is the base URL for your Domino server.
//   The mail service MUST be installed and enabled
//   on the server.  Be sure to change the protocol to
//   http if your server doesn't support https.
//
// - username is the name of a user who has access to a
//   mail file on your Domino server
//
// - password is the user's password
//
// - folder is the mail file folder name relative to the Domino
//   data directory.  Use '.' if the mail file is in the data
//   directory itself.
//
// - database is the mail file name.
//
var basePath = 'http://yourserver.yourorg.com';
var username = 'First Last';
var password = 'password';
var folder = 'mail';
var database = 'database.nsf';

/* Globals */

var view = 'inbox';

/* Callback for message list request */

var callback = function(error, data, response) {
  if (error) {
    console.error(error);
  } 
  else {
    if ( data != null || data.length == 0 ) {
      console.log('Request succeeded. Response follows ...\n');

      for ( var i = 0; i < data.length; i++ ) {
        console.log('Message ' + i);
        console.log('  subject: ' + data[i].subject);
        console.log('  date: ' + data[i].date);
        
        if ( view == "sent" || view == 'drafts' ) {
          console.log('  to: ' + data[i].to.displayName);
        }
        else {
          console.log('  from: ' + data[i].from.displayName);
        }

        console.log();
      }
    }
    else {
      console.log('Send mail request succeeded, but response does not contain a Location header.');
    }
  }
};

/* Start of main routine */

// Parse command line arguments

process.argv.forEach(function (val, index, array) {
  if ( index > 1 ) {
    view = val;
  }
});

// Get the API instance and set the base path of 
// the target Domino server

var api = new api.MessageListApi();
api.apiClient.basePath = basePath;

// Set user name and password

var basic = api.apiClient.authentications['basic'];
basic.username = username;
basic.password = password;

// Set up options

var opts = {
  'count': 5,
  'sortcolumn': 'date',
  'sortorder': 'descending'
};

// Send the request

if ( view == 'sent' ) {
  console.log('Reading sent view from ' + database + ' ...');
  api.folderDatabaseApiMailSentGet(folder, database, opts, callback);
}
else if ( view == 'drafts' ) {
  console.log('Reading drafts view from ' + database + ' ...');
  api.folderDatabaseApiMailDraftsGet(folder, database, opts, callback);
}
else if ( view == 'trash' ) {
  console.log('Reading trash view from ' + database + ' ...');
  api.folderDatabaseApiMailTrashGet(folder, database, opts, callback);
}
else if ( view == 'inbox' ) {
  console.log('Reading inbox from ' + database + ' ...');
  api.folderDatabaseApiMailInboxGet(folder, database, opts, callback);
}
else {
  console.log('Error.  Unknown view: ' + view);
}
